### PR TITLE
Fetch my user profile when my room member changed.

### LINKED ISSUE #1715: myroomnick changes the Global name
Hello,

I changed the nick using /myroomnick in a room, and Element-Android changed my global nick. Element on the mobile is showing the roomnick while Element-Web is correctly showing the previous global nick. 

Element version 1.0.0

Thanks

### PR DESCRIPTION
**Pull request by onurays**
Fixes #1715

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt [hunk 1]
```diff
@@ -22,9 +22,29 @@ import org.matrix.android.sdk.api.session.events.model.toModel
...
+import org.matrix.android.sdk.internal.di.SessionDatabase
+import org.matrix.android.sdk.internal.di.UserId
+import org.matrix.android.sdk.internal.session.profile.GetProfileInfoTask
+import org.matrix.android.sdk.internal.task.TaskExecutor
+import org.matrix.android.sdk.internal.task.configureWith
+import timber.log.Timber
 import javax.inject.Inject
 
-internal class RoomMemberEventHandler @Inject constructor() {
+internal class RoomMemberEventHandler @Inject constructor(
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt:40]: Why are you processing in this class and not in a EventInsertLiveProcessor?

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt [hunk 2]
```diff
@@ -41,10 +61,54 @@ internal class RoomMemberEventHandler @Inject constructor() {
...
+        if (roomMember.membership.isActive() && shouldUpdateUserEntity(realm, userId, roomMember)) {
+            updateUserEntity(realm, userId, roomMember)
+            if (userId == this.userId) {
+                fetchUserProfile(userId) // To fix #1715 (myroomnick changes the global name)
+            }
         }
         return true
     }
+
+    private fun shouldUpdateUserEntity(realm: Realm, userId: String, roomMember: RoomMemberContent?): Boolean {
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt:74]: You are fetching an entity to check for changes and then we use insertOrUpdate, so it's fetching again. 
Please re-use the UserEntity directly to update its fields

**Review by ganfra — CHANGES_REQUESTED** [review]

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt (hunk 1 above)
Comment (ganfra) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt:40]: You could also inherit from RealmLiveEntityObserver<UserEntity> and create a query for our userId

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt (hunk 2 above)
Comment (onurays) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt]: Fixed by reusing the entity, I want to discuss using RealmLiveEntityObserver.

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt [hunk 3]
```diff
@@ -41,10 +60,56 @@ internal class RoomMemberEventHandler @Inject constructor() {
...
+                }
+                ?: run {
+                    val userEntity = UserEntityFactory.create(userId, roomMember)
+                    realm.insertOrUpdate(userEntity)
+                    return true
+        }
+    }
+
+    private fun fetchUserProfile(userId: String) {
+        val params = GetProfileInfoTask.Params(userId)
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt:93]: If you let that here, could you create a task or use awaitCallback instead?

**Review by ganfra — CHANGES_REQUESTED** [review]

**Review by ganfra — CHANGES_REQUESTED** [review]
I guess it's ok for now

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt (hunk 3 above)
Comment (onurays) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt:93]: Sorry, I didn't understand your suggestion. Could you describe it a bit?

**Comment by bmarty** [comment]
This is not the ideal solution. See https://github.com/vector-im/element-android/pull/2816#issuecomment-778105195 for what we will do instead.

**Pull request closed without merging**